### PR TITLE
Fixed #18995 -- Made blocktranslate tag raise TemplateSyntaxError when plural count is not a number.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
ticket-18995

### CODE REVIEW

File: django/templatetags/i18n.py [hunk 1]
```diff
@@ -135,6 +135,8 @@ def render(self, context, nested=False):
         singular, vars = self.render_token_list(self.singular)
         if self.plural and self.countervar and self.counter:
             count = self.counter.resolve(context)
+            if not isinstance(count, int):
+                raise TemplateSyntaxError(f'"count" in \'{self.tag_name}\' tag must be an integer.')
```
Comment (felixxm) [inline django/templatetags/i18n.py:139]: As far as I'm aware `count` works fine with any numbers not only integers, see [comment](https://code.djangoproject.com/ticket/18995#comment:2):
```suggestion
            if not isinstance(count, (decimal.Decimal, float, int)):
                raise TemplateSyntaxError("'count' argument to %r tag must be a number.' % self.tag_name)
```

**Review by felixxm — COMMENTED** [review]
@jacobtylerwalls Thanks :+1:

File: django/templatetags/i18n.py (hunk 1 above)
Comment (jacobtylerwalls) [reply django/templatetags/i18n.py:139]: Thanks, good catch.

**Review by felixxm — COMMENTED** [review]
@jacobtylerwalls Thanks for updates :+1:

**Pull request merged**